### PR TITLE
Adding a Yosys+nextpnr+Trellis backend for the ECP5

### PR DESCRIPTION
**Pull request by daveshah1**

### CODE REVIEW

File: edalize/trellis.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+                elif f.file_type == 'user':
+                    pass
+            for key, value in self.vlogparam.items():
+                _s = "chparam -set {} {} $abstract\{}\n"
+                yosys_file.write(_s.format(key,
+                                           self._param_value_str(value, '"'),
+                                           self.toplevel))
+
+            yosys_file.write("verilog_defaults -pop\n")
+            yosys_file.write("synth_ecp5 -nomux")
```
Comment (olofk) [inline edalize/trellis.py]: Do we want -nomux by default, or is this something that the user should provide when required?

File: fusesoc/capi1/section.py [hunk 2]
```diff
@@ -490,6 +490,21 @@ def __init__(self, items=None):
...
+
+    TAG = 'trellis'
+
+    def __init__(self, items=None):
+        super(TrellisSection, self).__init__()
+
+        self._add_member('nextpnr_options', StringList, "nextpnr options")
+        self._add_member('yosys_synth_options' , StringList, "Additional options for the synth_* commands in yosys")
+        self._add_member('top_module', str, 'RTL top-level module')
+        self.top_module = 'orpsoc_top'
```
Comment (olofk) [inline fusesoc/capi1/section.py]: We don't want orpsoc_top as the default top_module. This is only a remain in some of the backends for backwards compatibility

File: edalize/trellis.py (hunk 1 above)
Comment (daveshah1) [reply edalize/trellis.py]: At the moment, and for the near future, nextpnr does not support the LUT extension muxes, so disabling them seems best for now

File: fusesoc/capi1/section.py (hunk 2 above)
Comment (daveshah1) [reply fusesoc/capi1/section.py]: Ah, think I added this while trying to get the old SoC project to build.

**Comment by olofk** [comment]
Picked and pushed. Thanks!

**Pull request closed without merging**